sdk/pinocchio/src/account_info.rs

```diff
@@ -610,6 +595,49 @@ impl AccountInfo {
     }
 }
 
+/// Mask to "consume" the mutable borrow bit for lamports.
+const UNSET_MUTABLE_LAMPORTS_BIT: u8 = 0b_0111_1111;
+
+/// Mask representing the immutable borrow bits for lamports.
+const IMMUTABLE_LAMPORTS_BITS: u8 = 0b_0111_0000;
+
+/// Mask representing the borrow bits for lamports.
+const LAMPORTS_BITS: u8 = 0b_1111_0000;
+
+/// Mask to assert the mutable borrow lamports bit.
+const MUTABLE_LAMPORTS_BORROW_BIT: u8 = 0b_1000_0000;
+
+/// Mask to "consume" the mutable borrow bit for data.
+const UNSET_MUTABLE_DATA_BIT: u8 = 0b_1111_0111;
+
+/// Mask representing the immutable borrow bits for data.
+const IMMUTABLE_DATA_BITS: u8 = 0b_0000_0111;
+
+/// Mask representing the borrow bits for data.
+const DATA_BITS: u8 = 0b_0000_1111;
+
+/// Mask to assert the mutable borrow data bit.
+const MUTABLE_DATA_BORRROW_BIT: u8 = 0b_0000_1000;
+
+/// Represents masks for borrow state of an account.
+#[repr(u8)]
+#[derive(Clone, Copy)]
+pub enum BorrowState {
+    /// Mask to check whether an account is already borrowed.
+    ///
+    /// This will test both data and lamports borrow state. Any position
+    /// in the borrow byte that is not set means that the account
+    /// is borrowed in that state.
+    Borrowed = 0b_1111_1111,
+
+    /// Mask to check whether an account is already mutably borrowed.
+    ///
+    /// This will test both data and lamports mutable borrow state. If
+    /// one of the mutably borrowed bits is not set, then the account
+    /// is mutably borrowed in that state.
+    MutablyBorrowed = 0b_1000_1000,
+}
```

Comment:
nit: Do you still need this with all of the consts? It might be more consistent to go with all one approach or the other, ie. all consts, or all enum variations